crates/deltalake-core/src/table/mod.rs

```diff
@@ -637,17 +638,31 @@ impl DeltaTable {
         &mut self,
         version: i64,
     ) -> Result<i64, DeltaTableError> {
+        // Use a cached timestamp info if available
         match self.version_timestamp.get(&version) {
             Some(ts) => Ok(*ts),
             None => {
-                let meta = self
-                    .object_store()
-                    .head(&commit_uri_from_version(version))
-                    .await?;
-                let ts = meta.last_modified.timestamp_millis();
+                // Load the version specified
+                if self.version() != version {
```

Comment:
Instead of loading the version each time, we could also just do this

`self.state.commit_infos().get(version).unwrap().timestamp`

Reply:
Just think aloud here but in general you shouldn't be able to restore to a version of a table that's above the current loaded table version